org.lflang/src/org/lflang/federated/generator/FedFileConfig.java

```diff
@@ -77,4 +78,10 @@ public Path getFedSrcPath() {
     public Path getFedSrcGenPath() {
         return getFedGenPath().resolve("src-gen");
     }
+
+    @Override
+    public void doClean() throws IOException {
+        super.doClean();
+        FileUtil.deleteDirectory(this.getFedSrcGenPath());
```

Comment:
As mentioned earlier, this line only removes the `src-gen` subdir. It's the parent of it that has to be deleted.